plugins/inputs/docker_log/client.go

```diff
@@ -0,0 +1,62 @@
+package docker_logs
+
+import (
+	"context"
+	"crypto/tls"
+	"net/http"
+
+	"github.com/docker/docker/api/types"
+	docker "github.com/docker/docker/client"
+	"github.com/docker/go-connections/sockets"
+	"io"
+)
+
+/*This file is inherited from telegraf docker input plugin*/
+var (
+	version        = "1.24"
+	defaultHeaders = map[string]string{"User-Agent": "engine-api-cli-1.0"}
+)
+
+type Client interface {
+	ContainerList(ctx context.Context, options types.ContainerListOptions) ([]types.Container, error)
+	ContainerLogs(ctx context.Context, containerID string, options types.ContainerLogsOptions) (io.ReadCloser, error)
+}
+
+func NewEnvClient() (Client, error) {
+	client, err := docker.NewEnvClient()
```

Comment:
Technically this is [deprecated](https://godoc.org/github.com/docker/docker/client#NewEnvClient) and should be `NewClientWithOpts(FromEnv)`

Reply:
Taken care of this!